src/status_im/keycard/real_keycard.cljs

```diff
@@ -321,7 +321,10 @@
      (let [{:keys [error]} (types/json->clj result)]
        (if-not (string/blank? error)
```

Comment:
if we start using the callbacks, we should not handle errors though on the on-success, or that's still an eventuality? Similarly in other parts of the code

Reply:
yes this will definitely need some more refactoring :) 
But since i did not write this functionality , I was hesitant on modifying it.. I will look at this one more time :)